File: golang/uPIMulator/src/linker/kernel/directive/quad_directive.go
package directive

import (
	"uPIMulator/src/abi/encoding"
	"uPIMulator/src/abi/word"
)

type QuadDirective struct {
	immediate *word.Immediate
}

func (this *QuadDirective) Init(value int64) {
	this.immediate = new(word.Immediate)
	this.immediate.Init(word.UNSIGNED, 8*int(this.Size()), value)
}

func (this *QuadDirective) Size() int64 {
	return 8
}

func (this *QuadDirective) Immediate() *word.Immediate {
	return this.immediate
}

func (this *QuadDirective) Encode() *encoding.ByteStream {
	return this.immediate.ToByteStream()
}

File: python_cpp/uPIMulator_frontend/src/abi/directive/quad_directive.py
from abi.word.immediate import Immediate
from abi.word.representation import Representation


class QuadDirective:
    def __init__(self, value: int):
        self._value: Immediate = Immediate(Representation.UNSIGNED, 8 * QuadDirective.size(), value)

    def value(self) -> int:
        return self._value.value()

    @staticmethod
    def size() -> int:
        return 8
